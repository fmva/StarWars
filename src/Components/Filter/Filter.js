import { Component } from 'react';
import React from 'react';
import { Badge, Button, Form, Input, Icon } from 'antd';
import PropTypes from 'prop-types';
import Mobile from '../../Utils/Mobile';
import Resources from '../../Resources/Resources';
import Utils from '../../Utils/Utils';

import './Filter.less';

class Filter extends Component {
  static propTypes = {
    /** show/hide filter fields for mobiles */
    mobileShowFilter: PropTypes.bool,
    /** header filter click handler */
    onClickMobileHeader: PropTypes.func,
    /** data for filter inputs */
    filterInputs: PropTypes.array,
    /** admitted data for filter inputs */
    filterAdmitted: PropTypes.array,
    /** admit button handler*/
    onClickAdmit: PropTypes.func,
    /** save filter input data to state*/
    saveFilterData: PropTypes.func,
    /** reset handler*/
    onClickReset: PropTypes.func
  };

  constructor(props) {
    super(props);
    this.handlerClickFilter = this.handlerClickFilter.bind(this);
    this.handlerFilterChange = this.handlerFilterChange.bind(this);
    this.handlerSubmitForm = this.handlerSubmitForm.bind(this);
    this.countFullInputs = this.countFullInputs.bind(this);
  }

  /**
   * handler for click of header filter
   */
  handlerClickFilter() {
    if (!Mobile.mobileCheck()) {
      return;
    }
    this.props.onClickMobileHeader();
  }

  /**
   * handler for submitting form
   * @param {Object} e
   */
  handlerSubmitForm(e) {
    e.preventDefault();
    this.props.onClickAdmit();
  }

  /**
   * render filter inputs
   * @param {Array} arr - array of inputs
   */
  renderInputs(arr) {
    const listInputs = arr.map((item, key) => (
      <Input
        autoComplete="off"
        autoCorrect="off"
        autoCapitalize="off"
        spellCheck="false"
        key={key}
        className="filter-input"
        size="large"
        placeholder={item.title}
        name={item.name}
        value={item.value}
        maxLength={60}
        data-id={key}
        onChange={this.handlerFilterChange}
      />
    ));

    return <div>{listInputs}</div>;
  }

  /**
   * check all inputs. If all inputs are empty the function return false
   */
  countFullInputs() {
    const filterInputs =
      this.props.filterInputs && this.props.filterInputs.length > 0 ? this.props.filterInputs : [];

    return filterInputs.reduce((previousValue, currentValue) => {
      return currentValue.value !== '' ? ++previousValue : previousValue;
    }, 0);
  }

  /**
   * handler for change filter
   */
  handlerFilterChange(evt) {
    this.props.saveFilterData(parseInt(evt.target.getAttribute('data-id'), 10), evt.target.value);
  }

  render() {
    const mobileShowFilter = this.props.mobileShowFilter ? this.props.mobileShowFilter : null;
    const filterInputs =
      this.props.filterInputs && this.props.filterInputs.length > 0 ? this.props.filterInputs : [];
    const onClickReset = this.props.onClickReset ? this.props.onClickReset : null;
    const filterAdmitted =
      this.props.filterAdmitted && this.props.filterAdmitted.length > 0
        ? this.props.filterAdmitted
        : [];

    return (
      <aside className={'filter-menu ' + (mobileShowFilter ? 'filter-menu-mobile-open' : '')}>
        <div className="filter-header-block">
          <div className="filter-header-title" onClick={this.handlerClickFilter}>
            <h2 className="filter-header">{Resources.getResource('filter_title')}</h2>
            <span className="phone-visible_inline">
              {mobileShowFilter ? (
                <Icon type="caret-up" theme="filled" />
              ) : (
                <Icon type="caret-down" theme="filled" />
              )}
            </span>
          </div>
          <div
            className={
              'filter-header-reset ' + (this.countFullInputs() === 0 ? '' : 'phone-visible-block')
            }
          >
            <Badge count={this.countFullInputs()}>
              <Button
                type="primary"
                shape="circle"
                icon="close"
                onClick={onClickReset}
                className="filter-button"
              />
            </Badge>
          </div>
        </div>
        <Form
          onSubmit={this.handlerSubmitForm}
          className={'filter-fields ' + (!mobileShowFilter ? 'phone-visible-none' : '')}
        >
          {this.renderInputs(filterInputs)}
          <div className="filter-button-block">
            <Button
              className="filter-button"
              type="primary"
              htmlType="submit"
              disabled={Utils.isEqualAdmittedFilter(filterInputs, filterAdmitted)}
            >
              {Resources.getResource('admit_button')}
            </Button>

            <Button
              onClick={onClickReset}
              className="filter-button phone-visible-inline-none"
              disabled={this.countFullInputs() === 0}
            >
              {Resources.getResource('reset_button')}
            </Button>
          </div>
        </Form>
      </aside>
    );
  }
}

export default Filter;
