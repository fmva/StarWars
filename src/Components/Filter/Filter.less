@import "../../Less/overall.less";

.filter-menu{
  border-left:1px solid #fff;
  width: 30%;
  padding:10px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-align-items: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  order:2;
  @media @phone {
    display: block;
    width: 100%;
    text-align: center;
    border:1px solid #bdcfda;
    border-radius: 3px;
    box-shadow: 0 0 6px #a0a2a9;

    position: fixed;
    z-index: 10;
    background-color: #f4f6f9;

  }
}

.filter-menu-mobile-open{
  @media @phone {
    height: calc(100% - @header-height);
    overflow: auto;
  }
}

.filter-fields{
  width: 90%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-align-items: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  @media @phone {
    margin: 0 auto;
  }
}

.filter-fields > div {
  width: 100%
}

.filter-input{
  width: 100%;
  margin:5px 0;
}

.filter-header{
  display: inline-block;
  padding: 2px 5px;
}

.filter-button{
  margin: 2px 3px;
}

.filter-button-block {
  text-align: center;
  @media @phone {
    width:80%;
  }
}

.filter-button-block button{
  @media @phone {
    width:100%;
  }
}

.filter-header-block {
  display: block;
  height:40px;
  @media @phone {
    cursor: pointer;
    position: relative;
  }
}

.filter-header-reset{
  display:none;
  @media @phone {
    position: absolute;
    right: 0;
    top: 0;
    z-index: 10;
  }
}

.filter-header-title {
  @media @phone {
    width:100%;
  }
}
