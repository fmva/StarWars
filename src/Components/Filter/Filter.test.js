import Filter from './Filter';
import React from 'react';
import { shallow, mount } from 'enzyme/build';
import Mobile from '../../Utils/Mobile';

const filter = [
  {
    title: 'title 1',
    value: '',
    name: 'cost_in_credits'
  },
  {
    title: 'title 2',
    value: '',
    name: 'manufacturer'
  },
  {
    title: 'title 3',
    value: 'value 3',
    name: 'starship_class'
  }
];

const filterAdmitted = [
  {
    title: 'title 1',
    value: '',
    name: 'cost_in_credits'
  },
  {
    title: 'title 2',
    value: '',
    name: 'manufacturer'
  },
  {
    title: 'title 3',
    value: 'value 3',
    name: 'starship_class'
  }
];

const onClickMobileHeader = jest.fn();
const onClickAdmit = jest.fn();
const onClickReset = jest.fn();
const saveFilterData = jest.fn();

let wrapper;

describe('Filter', () => {
  beforeAll(() => {
    wrapper = mount(
      <Filter
        mobileShowFilter={false}
        onClickMobileHeader={onClickMobileHeader}
        filterInputs={filter}
        filterAdmitted={filterAdmitted}
        onClickAdmit={onClickAdmit}
        onClickReset={onClickReset}
        saveFilterData={saveFilterData}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Компонент Filter c входными данными', () => {
    const wrapperShallow = shallow(
      <Filter
        mobileShowFilter={false}
        onClickMobileHeader={onClickMobileHeader}
        filterInputs={filter}
        filterAdmitted={filterAdmitted}
        onClickAdmit={onClickAdmit}
        onClickReset={onClickReset}
        saveFilterData={saveFilterData}
      />
    );
    expect(wrapperShallow).toMatchSnapshot();

    expect(wrapperShallow.find('Form.filter-fields Input').length).toBe(3);

    expect(
      wrapperShallow
        .find('Form.filter-fields Input')
        .at(0)
        .props().placeholder
    ).toBe('title 1');
    expect(
      wrapperShallow
        .find('Form.filter-fields Input')
        .at(1)
        .props().placeholder
    ).toBe('title 2');
    expect(
      wrapperShallow
        .find('Form.filter-fields Input')
        .at(2)
        .props().placeholder
    ).toBe('title 3');
    expect(
      wrapperShallow
        .find('Form.filter-fields Input')
        .at(2)
        .props().value
    ).toBe('value 3');
  });

  it(
    'Допустим пользователь нажимает на заголовок в мобильном телефоне, ' +
      'должно сработать событие onClickMobileHeader',
    () => {
      Mobile.mobileCheck = jest.fn();
      Mobile.mobileCheck.mockReturnValue(true);

      wrapper.find('div.filter-header-title').simulate('click');
      expect(onClickMobileHeader.mock.calls.length).toBe(1);

      Mobile.mobileCheck.mockReset();
      onClickMobileHeader.mockClear();
    }
  );

  it(
    'Допустим пользователь нажимает на заголовок в десктопном приложении, ' +
      'событие onClickMobileHeader не должно сработать',
    () => {
      Mobile.mobileCheck = jest.fn();
      Mobile.mobileCheck.mockReturnValue(false);

      wrapper.find('div.filter-header-title').simulate('click');
      expect(onClickMobileHeader.mock.calls.length).toBe(0);

      Mobile.mobileCheck.mockReset();
      onClickMobileHeader.mockClear();
    }
  );

  it(
    'Допустим пользователь нажимает на кнопку применить фильтр, ' +
      'событие onClickAdmit должно сработать',
    () => {
      wrapper.find('form.filter-fields').simulate('submit');
      expect(onClickAdmit.mock.calls.length).toBe(1);

      onClickAdmit.mockClear();
    }
  );

  it(
    'Допустим пользователь нажимает на кнопку отменить фильтр, ' +
      'событие onClickReset должно сработать',
    () => {
      wrapper
        .find('div.filter-button-block button')
        .at(1)
        .simulate('click');
      expect(onClickReset.mock.calls.length).toBe(1);

      onClickReset.mockClear();
    }
  );

  it(
    'Допустим пользователь вводит данные в поля формы, ' +
      'событие saveFilterData должно сработать',
    () => {
      wrapper
        .find('form.filter-fields input')
        .at(0)
        .simulate('change');
      expect(saveFilterData.mock.calls.length).toBe(1);

      saveFilterData.mockClear();
    }
  );
});
