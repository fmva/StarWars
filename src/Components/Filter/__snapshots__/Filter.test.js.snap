// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Filter
  filterAdmitted={
    Array [
      Object {
        "name": "cost_in_credits",
        "title": "title 1",
        "value": "",
      },
      Object {
        "name": "manufacturer",
        "title": "title 2",
        "value": "",
      },
      Object {
        "name": "starship_class",
        "title": "title 3",
        "value": "value 3",
      },
    ]
  }
  filterInputs={
    Array [
      Object {
        "name": "cost_in_credits",
        "title": "title 1",
        "value": "",
      },
      Object {
        "name": "manufacturer",
        "title": "title 2",
        "value": "",
      },
      Object {
        "name": "starship_class",
        "title": "title 3",
        "value": "value 3",
      },
    ]
  }
  mobileShowFilter={false}
  onClickAdmit={[MockFunction]}
  onClickMobileHeader={[MockFunction]}
  onClickReset={[MockFunction]}
  saveFilterData={[MockFunction]}
>
  <aside
    className="filter-menu "
  >
    <div
      className="filter-header-block"
    >
      <div
        className="filter-header-title"
        onClick={[Function]}
      >
        <h2
          className="filter-header"
        >
          Filter
        </h2>
        <span
          className="phone-visible_inline"
        >
          <Icon
            theme="filled"
            type="caret-down"
          >
            <LocaleReceiver
              componentName="Icon"
            >
              <i
                aria-label="icon: caret-down"
                className="anticon anticon-caret-down"
              >
                <IconReact
                  className=""
                  type="caret-down-fill"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="caret-down"
                    fill="currentColor"
                    height="1em"
                    key="svg-caret-down"
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                      key="svg-caret-down-svg-0"
                    />
                  </svg>
                </IconReact>
              </i>
            </LocaleReceiver>
          </Icon>
        </span>
      </div>
      <div
        className="filter-header-reset phone-visible-block"
      >
        <Badge
          count={1}
          dot={false}
          overflowCount={99}
          showZero={false}
        >
          <span
            className="ant-badge"
          >
            <Button
              block={false}
              className="filter-button"
              ghost={false}
              icon="close"
              loading={false}
              onClick={[MockFunction]}
              shape="circle"
              type="primary"
            >
              <Wave>
                <button
                  className="ant-btn filter-button ant-btn-primary ant-btn-circle ant-btn-icon-only"
                  onClick={[Function]}
                  type="button"
                >
                  <Icon
                    type="close"
                  >
                    <LocaleReceiver
                      componentName="Icon"
                    >
                      <i
                        aria-label="icon: close"
                        className="anticon anticon-close"
                      >
                        <IconReact
                          className=""
                          type="close-o"
                        >
                          <svg
                            aria-hidden="true"
                            className=""
                            data-icon="close"
                            fill="currentColor"
                            height="1em"
                            key="svg-close"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                              key="svg-close-svg-0"
                            />
                          </svg>
                        </IconReact>
                      </i>
                    </LocaleReceiver>
                  </Icon>
                </button>
              </Wave>
            </Button>
            <Animate
              animation={Object {}}
              component=""
              componentProps={Object {}}
              onAppear={[Function]}
              onEnd={[Function]}
              onEnter={[Function]}
              onLeave={[Function]}
              showProp="data-show"
              transitionAppear={true}
              transitionEnter={true}
              transitionLeave={true}
              transitionName="ant-badge-zoom"
            >
              <AnimateChild
                animation={Object {}}
                key="scrollNumber"
                transitionAppear={true}
                transitionEnter={true}
                transitionLeave={true}
                transitionName="ant-badge-zoom"
              >
                <ScrollNumber
                  className="ant-badge-count"
                  count={1}
                  data-show={true}
                  key="scrollNumber"
                  onAnimated={[Function]}
                  prefixCls="ant-scroll-number"
                  title={1}
                >
                  <sup
                    className="ant-scroll-number ant-badge-count"
                    data-show={true}
                    title={1}
                  >
                    <span
                      className="ant-scroll-number-only"
                      key="0"
                      style={
                        Object {
                          "WebkitTransform": "translateY(-1100%)",
                          "msTransform": "translateY(-1100%)",
                          "transform": "translateY(-1100%)",
                          "transition": "none",
                        }
                      }
                    >
                      <p
                        className=""
                        key="0"
                      >
                        0
                      </p>
                      <p
                        className=""
                        key="1"
                      >
                        1
                      </p>
                      <p
                        className=""
                        key="2"
                      >
                        2
                      </p>
                      <p
                        className=""
                        key="3"
                      >
                        3
                      </p>
                      <p
                        className=""
                        key="4"
                      >
                        4
                      </p>
                      <p
                        className=""
                        key="5"
                      >
                        5
                      </p>
                      <p
                        className=""
                        key="6"
                      >
                        6
                      </p>
                      <p
                        className=""
                        key="7"
                      >
                        7
                      </p>
                      <p
                        className=""
                        key="8"
                      >
                        8
                      </p>
                      <p
                        className=""
                        key="9"
                      >
                        9
                      </p>
                      <p
                        className=""
                        key="10"
                      >
                        0
                      </p>
                      <p
                        className="current"
                        key="11"
                      >
                        1
                      </p>
                      <p
                        className=""
                        key="12"
                      >
                        2
                      </p>
                      <p
                        className=""
                        key="13"
                      >
                        3
                      </p>
                      <p
                        className=""
                        key="14"
                      >
                        4
                      </p>
                      <p
                        className=""
                        key="15"
                      >
                        5
                      </p>
                      <p
                        className=""
                        key="16"
                      >
                        6
                      </p>
                      <p
                        className=""
                        key="17"
                      >
                        7
                      </p>
                      <p
                        className=""
                        key="18"
                      >
                        8
                      </p>
                      <p
                        className=""
                        key="19"
                      >
                        9
                      </p>
                      <p
                        className=""
                        key="20"
                      >
                        0
                      </p>
                      <p
                        className=""
                        key="21"
                      >
                        1
                      </p>
                      <p
                        className=""
                        key="22"
                      >
                        2
                      </p>
                      <p
                        className=""
                        key="23"
                      >
                        3
                      </p>
                      <p
                        className=""
                        key="24"
                      >
                        4
                      </p>
                      <p
                        className=""
                        key="25"
                      >
                        5
                      </p>
                      <p
                        className=""
                        key="26"
                      >
                        6
                      </p>
                      <p
                        className=""
                        key="27"
                      >
                        7
                      </p>
                      <p
                        className=""
                        key="28"
                      >
                        8
                      </p>
                      <p
                        className=""
                        key="29"
                      >
                        9
                      </p>
                    </span>
                  </sup>
                </ScrollNumber>
              </AnimateChild>
            </Animate>
          </span>
        </Badge>
      </div>
    </div>
    <Form
      className="filter-fields phone-visible-none"
      hideRequiredMark={false}
      layout="horizontal"
      onSubmit={[Function]}
    >
      <form
        className="ant-form ant-form-horizontal filter-fields phone-visible-none"
        onSubmit={[Function]}
      >
        <div>
          <Input
            autoCapitalize="off"
            autoComplete="off"
            autoCorrect="off"
            className="filter-input"
            data-id={0}
            disabled={false}
            key="0"
            maxLength={60}
            name="cost_in_credits"
            onChange={[Function]}
            placeholder="title 1"
            size="large"
            spellCheck="false"
            type="text"
            value=""
          >
            <input
              autoCapitalize="off"
              autoComplete="off"
              autoCorrect="off"
              className="ant-input ant-input-lg filter-input"
              data-id={0}
              disabled={false}
              maxLength={60}
              name="cost_in_credits"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="title 1"
              size="large"
              spellCheck="false"
              type="text"
              value=""
            />
          </Input>
          <Input
            autoCapitalize="off"
            autoComplete="off"
            autoCorrect="off"
            className="filter-input"
            data-id={1}
            disabled={false}
            key="1"
            maxLength={60}
            name="manufacturer"
            onChange={[Function]}
            placeholder="title 2"
            size="large"
            spellCheck="false"
            type="text"
            value=""
          >
            <input
              autoCapitalize="off"
              autoComplete="off"
              autoCorrect="off"
              className="ant-input ant-input-lg filter-input"
              data-id={1}
              disabled={false}
              maxLength={60}
              name="manufacturer"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="title 2"
              size="large"
              spellCheck="false"
              type="text"
              value=""
            />
          </Input>
          <Input
            autoCapitalize="off"
            autoComplete="off"
            autoCorrect="off"
            className="filter-input"
            data-id={2}
            disabled={false}
            key="2"
            maxLength={60}
            name="starship_class"
            onChange={[Function]}
            placeholder="title 3"
            size="large"
            spellCheck="false"
            type="text"
            value="value 3"
          >
            <input
              autoCapitalize="off"
              autoComplete="off"
              autoCorrect="off"
              className="ant-input ant-input-lg filter-input"
              data-id={2}
              disabled={false}
              maxLength={60}
              name="starship_class"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="title 3"
              size="large"
              spellCheck="false"
              type="text"
              value="value 3"
            />
          </Input>
        </div>
        <div
          className="filter-button-block"
        >
          <Button
            block={false}
            className="filter-button"
            disabled={true}
            ghost={false}
            htmlType="submit"
            loading={false}
            type="primary"
          >
            <Wave>
              <button
                className="ant-btn filter-button ant-btn-primary"
                disabled={true}
                onClick={[Function]}
                type="submit"
              >
                <span
                  key=".0"
                >
                  Admit
                </span>
              </button>
            </Wave>
          </Button>
          <Button
            block={false}
            className="filter-button phone-visible-inline-none"
            disabled={false}
            ghost={false}
            loading={false}
            onClick={[MockFunction]}
          >
            <Wave>
              <button
                className="ant-btn filter-button phone-visible-inline-none"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span
                  key=".0"
                >
                  Reset
                </span>
              </button>
            </Wave>
          </Button>
        </div>
      </form>
    </Form>
  </aside>
</Filter>
`;

exports[`Filter Компонент Filter c входными данными 1`] = `
<aside
  className="filter-menu "
>
  <div
    className="filter-header-block"
  >
    <div
      className="filter-header-title"
      onClick={[Function]}
    >
      <h2
        className="filter-header"
      >
        Filter
      </h2>
      <span
        className="phone-visible_inline"
      >
        <Icon
          theme="filled"
          type="caret-down"
        />
      </span>
    </div>
    <div
      className="filter-header-reset phone-visible-block"
    >
      <Badge
        count={1}
        dot={false}
        overflowCount={99}
        showZero={false}
      >
        <Button
          block={false}
          className="filter-button"
          ghost={false}
          icon="close"
          loading={false}
          onClick={[MockFunction]}
          shape="circle"
          type="primary"
        />
      </Badge>
    </div>
  </div>
  <Form
    className="filter-fields phone-visible-none"
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
  >
    <div>
      <Input
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        className="filter-input"
        data-id={0}
        disabled={false}
        key="0"
        maxLength={60}
        name="cost_in_credits"
        onChange={[Function]}
        placeholder="title 1"
        size="large"
        spellCheck="false"
        type="text"
        value=""
      />
      <Input
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        className="filter-input"
        data-id={1}
        disabled={false}
        key="1"
        maxLength={60}
        name="manufacturer"
        onChange={[Function]}
        placeholder="title 2"
        size="large"
        spellCheck="false"
        type="text"
        value=""
      />
      <Input
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        className="filter-input"
        data-id={2}
        disabled={false}
        key="2"
        maxLength={60}
        name="starship_class"
        onChange={[Function]}
        placeholder="title 3"
        size="large"
        spellCheck="false"
        type="text"
        value="value 3"
      />
    </div>
    <div
      className="filter-button-block"
    >
      <Button
        block={false}
        className="filter-button"
        disabled={true}
        ghost={false}
        htmlType="submit"
        loading={false}
        type="primary"
      >
        Admit
      </Button>
      <Button
        block={false}
        className="filter-button phone-visible-inline-none"
        disabled={false}
        ghost={false}
        loading={false}
        onClick={[MockFunction]}
      >
        Reset
      </Button>
    </div>
  </Form>
</aside>
`;
